examples/with-app-directory/src/app/[lang]/layout.tsx

```diff
@@ -1,10 +1,16 @@
+import i18n from '../../../i18n'
+import { redirect } from 'next/navigation'
+
 export default function RootLayout({
   children,
   params: { lang },
 }: {
   children: React.ReactNode
   params: { lang: string }
 }) {
+  // Redirect to default locale if lang is not supported. /second-page -> /en/second-page
+  if (!i18n.locales.includes(lang)) redirect(`/${i18n.defaultLocale}/${lang}`)
```

Comment:
Hi there! I am a bit confused by this. Isn't this a redirect to sth. like `/en/de`, completely discarding the original pathname (which wouldn't be available in a layout in the first place to my understanding)? Also quite sure `/second-page -> /en/second-page` is not true, as it couldn't be redirecting a non-locale-prefixed path, as itself is located beneath the `[lang]` parameter and thus can't even catch `/second-page`.

Reply:
Being a dynamic path `/[lang]/about-us`, if someone enters without the language (`/about-us`), the "`about-us`" in this case would be the page although it has been taken as `lang`, but it is **not a correct `lang`**, so it redirects from `/about-us` to `/en/about-us`. This is the idea. It can probably be improved, but I encourage you to try the example and load the `/second-page` without lang.